backend/spec/controllers/spree/admin/payment_methods_controller_spec.rb

```diff
@@ -70,5 +70,33 @@ class GatewayWithPassword < PaymentMethod
         expect(assigns(:payment_methods).to_a).to eql([second_method, first_method])
       end
     end
+
+    describe "#update" do
+      # Regression test for https://github.com/solidusio/solidus/issues/2789
+      let(:payment_method) { create(:store_credit_payment_method) }
+      let(:params) do
+        {
+          id: payment_method.id,
+          payment_method: {
+            name: 'Check',
+            type: 'Spree::PaymentMethod::Check'
+          }
+        }
+      end
+
+      subject { put :update, params: params }
+
+      it 'redirects to payment method edit page' do
+        subject
+        expect(response).to redirect_to(spree.edit_admin_payment_method_path(payment_method))
+      end
+
+      it 'updates the resource' do
+        subject
+        Spree::PaymentMethod.all.reload
```

Comment:
I'm reloading the payment methods because I got this error if I use the `payment_method` created before:

```
pry(#<RSpec::ExampleGroups::SpreeAdminPaymentMethodsController::Update>)> payment_method.reload
ActiveRecord::RecordNotFound: Couldn't find Spree::PaymentMethod::StoreCredit with 'id'=1 [WHERE "spree_payment_methods"."type" IN ('Spree::PaymentMethod::StoreCredit')]
from /Users/kaifan11/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/activerecord-5.2.0/lib/active_record/relation/finder_methods.rb:346:in `raise_record_not_found_exception!'
```

I think it's because of the payment method reference, any ideas?

Reply:
That's weird. I don't think you're using the reference anymore though.